ouroboros-consensus/docs/Contributing.md

```diff
@@ -1,5 +1,35 @@
-TODO eg git history goals etc
+# How to contribute to the Consensus Layer
 
-See the
-https://github.com/input-output-hk/ouroboros-network/ouroboros-consensus/docs/StyleGuide.md
-document.
+See the follow documents.
+
+  * this repository's [README][repo-README]; it links to design documents
+  * Consensus [style guide][consensus-style]
+  * Consensus [git process][consensus-git]
+  * Consensus [onboarding][consensus-onboarding]
+  * TODO quick-start guide
+
+Our primary neighbors are the upstream [cardano-ledger-specs] and the downstream
+[cardano-node].
+
+If you're not an IOHK employee, please fork the repository on GitHub and open
+PRs that way. We're excited about your involvement; thank you!
+
+If you're an IOHK employee, work with your manager and DevOps to acquire
+permissions to push branches to and open PRs in this repository. If you're
+working as part of the Consensus Team, also acquire permissions to Approve PRs
+(in particular, our `bors` bot must accept your commands). You'll eventually
+also need to be in the `CODEOWNERS` file. The general IOHK onboarding should
+have discussed establishing hardware-based Two Factor Authentication on your
+GitHub account; that's another prerequisite.
+
+[repo-README]: https://github.com/input-output-hk/ouroboros-network/README.md
```

Comment:
Maybe we should not build cycles? I think README.md should be an entry point to the rest of the documentation, we should point from README to Contributing, not the other way around.

Reply:
Sounds good. Done.